File: week1/pset/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void row(int space, int brick);
int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1);

    for (int i = 0; i < height; i++)
    {
        row(height - (i + 1), i + 1);
        printf("\n");
    }
}

void brick_f(int number_of_bricks);
void row(int space, int brick)
{
    for (int i = 0; i < space; i++)
    {
        printf(" ");
    }
    brick_f(brick);
    printf("  ");
    brick_f(brick);
}

void brick_f(int number_of_bricks)
{
    for (int j = 0; j < number_of_bricks; j++)
    {
        printf("#");
    }
}

File: week6/pset/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break
    for i in range(height):
        row(height - (i + 1), i + 1)
        print("")


def row(space, brick):
    for i in range(space):
        print(" ", end="")
    brick_f(brick)
    print("  ", end="")
    brick_f(brick)


def brick_f(number_of_bricks):
    for i in range(number_of_bricks):
        print("#", end="")


main()
